BlueprintUI/Sources/Internal/CacheTree.swift

```diff
@@ -22,12 +25,31 @@ protocol CacheTree: AnyObject {
 extension CacheTree {
     /// Convenience method to get a cached size, or compute and store one if it is not in the cache.
     func get(_ constraint: SizeConstraint, orStore calculation: (SizeConstraint) -> CGSize) -> CGSize {
-        if let size = self[constraint] {
+
+        /// Due to various math upstream, we can end up with measurements like 210.0 getting
+        /// turned into numbers like 209.99999997 due to loss of precision.
+        ///
+        /// Because we cache by exact values in this cache,
+        /// let's round this up to the nearest screen-scale pixel.
+
+        let roundedConstraint = constraint.roundToNearestPixel(with: screenScale)
+
+        // TODO: Why does this break??
```

Comment:
This breaks one test; which is interesting; I _think_ because test elements are relying on their measurement constraint to be unrounded (even unrounded off of screen pixel values, eg 0.333). This feels "wrong", but didn't change it.